src/components/AppMap/AppMap.vue

```diff
@@ -96,6 +97,9 @@ export default {
       const boundingBox = bbox(this.selectedFeature);
       this.$root.map.fitBounds(boundingBox, { padding: 80 });
     },
+    updateStyle(style) {
+      this.mapConfig.style = style;
```

Comment:
I fixed a bug on `vue2mapbox-gl` now this is enough to change the style and to reload the layers